packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -953,16 +959,20 @@ function completeWork(
           // If new content unsuspended, but there's still some content that
           // didn't. Then we need to do a second pass that forces everything
           // to keep showing their fallbacks.
-          const needsRerender = hasSuspendedChildrenAndNewContent(
-            workInProgress,
-            renderedChildren,
-          );
+
+          // We might be suspended if something in this render pass suspended, or
+          // something in the previous committed pass suspended. Otherwise,
+          // there's no chance so we can skip the expensive call to
+          // hasSuspendedChildrenAndNewContent.
+          let cannotBeSuspended =
```

Comment:
This seems like something we could put on the `effectTag` or a similar bitfield that gets bubbled up during complete/unwind.

Reply:
Yea I wanted a tag on each child so that I can use it to efficiently determine the tail in beginWork.